astropy/coordinates/transformations.py

```diff
@@ -674,6 +674,108 @@ def deco(func):
             return func
         return deco
 
+    def _add_merged_transform(self, fromsys, viasys, tosys, priority=1):
```

Comment:
I really like this function, but having just one intermediate frame feels an unnecessary limitation. Perhaps for followup, but 

```suggestion
    def _add_merged_transform(self, fromsys, viasys, *tosys, priority=1):
```

And you use the same recursion trick as ``_get_equivalent_affine_params_to``

Reply:
Suggested functionality now implemented (although not via recursion)